substratools/workspace.py

```diff
@@ -56,84 +54,33 @@ def __init__(
             DEFAULT_INPUT_DATA_FOLDER_PATH
         )
 
-        self.input_predictions_path = input_predictions_path or self._get_default_path(DEFAULT_INPUT_PREDICTIONS_PATH)
-
-        self.output_predictions_path = output_predictions_path or self._get_default_path(
-            DEFAULT_OUTPUT_PREDICTIONS_PATH
-        )
-
-        dirs = []
-        dirs.extend(self.input_data_folder_paths)
-        paths = [
-            self.input_predictions_path,
-            self.output_predictions_path,
-        ]
-        dirs.extend([os.path.dirname(p) for p in paths])
-        for d in dirs:
+        for d in self.input_data_folder_paths:
             if d:
                 makedir_safe(d)
 
 
-class MetricsWorkspace(OpenerWorkspace):
-    """Filesystem workspace for metrics execution."""
+class GenericAlgoWorkspace(OpenerWorkspace):
+    """Filesystem workspace for user defined method execution."""
 
     def __init__(
         self,
         dirpath=None,
-        input_data_folder_paths=None,
-        input_predictions_path=None,
-        output_perf_path=None,
-        log_path=None,
-    ):
-        super().__init__(
-            dirpath=dirpath,
-            input_data_folder_paths=input_data_folder_paths,
-            input_predictions_path=input_predictions_path,
-        )
-
-        self.output_perf_path = output_perf_path or self._get_default_path(DEFAULT_OUTPUT_PERF_PATH)
-
-        self.log_path = log_path or self._get_default_path(DEFAULT_LOG_PATH)
-
-        dirs = []
-        paths = [
-            self.output_perf_path,
-            self.log_path,
-        ]
-        dirs.extend([os.path.dirname(p) for p in paths])
-        for d in dirs:
-            if d:
-                makedir_safe(d)
-
-
-class AlgoWorkspace(OpenerWorkspace):
-    """Filesystem workspace for algo execution."""
-
-    def __init__(
-        self,
-        dirpath=None,
-        input_data_folder_paths=None,
-        input_model_paths=None,
-        input_predictions_path=None,
-        output_model_path=None,
-        output_predictions_path=None,
         log_path=None,
         chainkeys_path=None,
+        inputs=None,
+        outputs=None,
     ):
-        super().__init__(
-            dirpath=dirpath,
-            input_data_folder_paths=input_data_folder_paths,
-            input_predictions_path=input_predictions_path,
-            output_predictions_path=output_predictions_path,
-        )
 
-        self.input_model_paths = input_model_paths
-        self.output_model_path = output_model_path
+        super().__init__(dirpath, inputs.input_data_folder_paths) if inputs is not None else super().__init__(dirpath)
```

Comment:
```suggestion
        super().__init__(dirpath=dirpath, input_data_folder_paths=None if inputs is None else inputs.input_data_folder_paths)
```

minor suggestion to put only the value of the `input_data_folder_paths` arg in the ternary

Reply:
done